### ISSUE TITLE
walrus operator in chain expression compiles to invalid python

### ISSUE THREAD

**Issue by bj0** [evhub/coconut#519]
I tried to do something similar to this:
```
def some_expensive_function(a) = a*10

def f(a) = (c:=some_expensive_function(a)) :: f(c)
```

but it compiles to code which throws:
>SyntaxError: assignment expression cannot be used in a comprehension iterable expression

**Comment by evhub** [comment]
https://github.com/evhub/coconut/commit/ddd717a3de1460958442fe43dff16668e846efc6 (`coconut-develop>=1.5.0-post_dev31`) fixes this for `::`, though the simple `(x for x in (y:= z))` still produces this `SyntaxError` on `--target 3.8`.

**Comment by evhub** [comment]
Fully resolved in `coconut-develop>=1.5.0-post_dev32`.
